src/pages/login/index.js

```diff
@@ -8,10 +8,39 @@ import './login.css';
 const Login = () => {
   const { onLogin } = useAuth();
   const [formData, setFormData] = useState({ email: '', password: '' });
+  const [emailIsValid, setEmailIsValid] = useState(false);
+  const [passwordIsValid, setPassordIsValid] = useState(false);
+
+  const isValidEmail = () => {
+    // validates abc@de.fh as email. must be 3 characters on both sides of @ and no trailing dot
+    const validRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    setEmailIsValid(validRegex.test(formData.email));
+    console.log('Valid: ', validRegex.test(formData.email), '. Email: ', formData.email);
+    return validRegex.test(formData.email);
+  };
+
+  const isValidPassword = () => {
```

Comment:
this function is repeated twice, once in login and once in registration. Please move it into an utils.js file and call it something like:

```js
validatePassword(password) { // name is indicative, you can choose a different name
....
}
```

The function should return true/false.